### PR TITLE
feat: Adds VitEncoder for visual transformers

### PR DESCRIPTION
**Pull request by bruvduroiu**
TODO:

- [x] formalise Pydantic types for routes
- [x] more unit tests that cover torch
- [x] documentation

### CODE REVIEW

File: semantic_router/encoders/vit.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    def _process_images(self, images: List[_Image]):
+        rgb_images = [self._ensure_rgb(img) for img in images]
+        processed_images = self._processor(images=rgb_images, return_tensors="pt")
+        processed_images = processed_images.to(self.device)
+        return processed_images
+
+    def _ensure_rgb(self, img: _Image):
+        rgbimg = Image.new("RGB", img.size)
+        rgbimg.paste(img)
+        return rgbimg
```
Comment (bruvduroiu) [inline semantic_router/encoders/vit.py:74]: Might want to have this as a utility when we integrate CLIP

File: semantic_router/encoders/vit.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                "You can install it with: "
+                "`pip install semantic-router[local]`"
+            )
+
+        self._torch = torch
+        self._T = T
+
+        processor = ViTImageProcessor.from_pretrained(
+            self.name, **self.processor_kwargs
+        )
```
Comment (bruvduroiu) [inline semantic_router/encoders/vit.py:51]: The `ViTImageProcessor` abstracts away the image preprocessing to bring any image to the 224x224 shape needed for ViT

File: semantic_router/encoders/vit.py (hunk 1 above)
Comment (jamescalam) [reply semantic_router/encoders/vit.py:74]: nice idea, after merging this to main I will merge it into the clip branch and move

File: semantic_router/utils/splitters.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
+from typing import Any, List, Literal, Optional
```
Comment (jamescalam) [inline semantic_router/utils/splitters.py:1]: this file `/utils/splitters.py` can be removed as it now exists in the `/splitters` module

**Review by jamescalam — APPROVED** [review]

**Pull request merged**